openapi_operations.go

```diff
@@ -86,6 +86,18 @@ func (r Route[ResponseBody, RequestBody]) Param(paramType ParamType, name, descr
 	return r
 }
 
+// Security adds a security requirement to the route.
+// It configures the route to require authentication using the specified security scheme.
+// The requirementName parameter should match a security scheme defined in the OpenAPI components section.
+// For example, if you have a JWT bearer auth scheme named "bearerAuth" in your components,
+// you would call Security("bearerAuth") to require JWT authentication for this route.
+func (r Route[ResponseBody, RequestBody]) Security(requirementName string) Route[ResponseBody, RequestBody] {
```

Comment:
This builder style pattern is ultimately deprecated. I don't think we'd want to add a new feature here. Would you mind implementing this in the style that is `option.go` [here is an example](https://github.com/go-fuego/fuego/blob/main/option.go#L204) and it's subsequent entry into the [`option` package](https://github.com/go-fuego/fuego/blob/main/option/option.go#L79)

Reply:
Yes, that's the only comment I have too. Thanks for making this PR!

Reply:
Thanks for the hint, sorry for the mistake, I am new to golang and also to fuego.
I explored security schemes and included the ability to combine multiple requirements to get the ability to put them in and / or with each other and also to assign them a scope. 

### Support for Single Security Requirements:
Allows adding a single security scheme, e.g., `basic` authentication:
```go
fuego.OptionSecurity(openapi3.SecurityRequirement{
    "basic": []string{},
})
```
### Support for Scopes:
Handles schemes with specific **scopes**, like `oauth2`:
```go
fuego.OptionSecurity(openapi3.SecurityRequirement{
    "oauth2": []string{"read:users", "write:users"},
})
```
### AND Logic Within a Single Requirement:

Combines multiple schemes in one `SecurityRequirement` for **AND** logic:
```go
fuego.OptionSecurity(openapi3.SecurityRequirement{
    "basic":  []string{},
    "oauth2": []string{"read:users"},
})
```
### OR Logic Across Multiple Requirements:

Adds multiple SecurityRequirement entries for **OR** logic:
```go
fuego.OptionSecurity(
    openapi3.SecurityRequirement{"basic": []string{}},
    openapi3.SecurityRequirement{"oauth2": []string{"read:users"}},
)
```
### Handles Complex Combinations:

Supports both **AND** and **OR** in the same configuration:
```go
fuego.OptionSecurity(
    openapi3.SecurityRequirement{
        "basic":  []string{},
        "oauth2": []string{"read:users"},
    },
    openapi3.SecurityRequirement{
        "apiKey": []string{},
    },
)
```